### PR TITLE
rspec: Avoid home projects for users created in factories

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: src/api/spec/factories/users.rb [hunk 1]
```diff
@@ -5,10 +5,29 @@
...
     end
 
+    trait :with_home do
+      avoid_home_project { false }
+    end
+
+    to_create do |user, evaluator|
+      before = Configuration.allow_user_to_create_home_project
+      if evaluator.avoid_home_project
+        Configuration.update(allow_user_to_create_home_project: false)
```
Comment (hennevogel) [inline src/api/spec/factories/users.rb]: Why don't you make this default in the suite setup, instead of changing it here all the time?

Comment (coolo) [reply src/api/spec/factories/users.rb]: You can still have 2 users in the same test - one with and one without home.

Comment (hennevogel) [reply src/api/spec/factories/users.rb]: Sure and the one `:with_home` can change the default during creation, no?

**Comment by vpereira** [comment]
@coolo  can you rebase with master and fix it here too? https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/trigger_controller_spec.rb

**Comment by coolo** [comment]
@hennevogel ping

**Comment by hennevogel** [comment]
@coolo sorry, a bit busy with other stuff. maybe later today

**Comment by hennevogel** [comment]
@coolo[bot] rebase

**Comment by coolo** [comment]
I can rebase!

**Pull request merged**